Issue-143: Add list comment support on frontend

Thinbishop opened the pull request:
## Issue
<!--
Automatically close the issue when this PR is merged
    USAGE: Fixes/Closes #<issue number>
-->
Closes #143 

## Type of Change
<!-- bug fix, feature, documentation, UI, etc. -->

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
<!-- Please add a detailed description of what this PR does and why it is needed -->
This PR adds list comment support to the frontend. Movie list pages will contain a comments section below the list contents. These comments will have:
- User Avatar (these are only fetched once per unique user even if they have multiple comments)
- Username
- Date/Time
- Whether the comment has been edited
- Comment text
- Editor functionality
- Delete comment button

An add comment input will appear at the top of the section, and any new comments will appear directly below (sorted descending by creation date).

The comment list will appear as a trimmed section (3 comments), providing a "Load More" button to expand the section to all comments.

The user context is used to ensure edit/delete can only be done by the user that created the comment.

Example from user 1
![image](https://github.com/chrispinkney/cinesync/assets/75338069/fe711050-7048-48a1-8184-8450cf9d9a82)

Example from user 2
![image](https://github.com/chrispinkney/cinesync/assets/75338069/a7bf8377-6521-4447-a396-f6fb081b6f31)

This PR also fixes avatar fetch.

## Testing the PR
<!-- Please add steps to build the changes in your PR locally so that your PR can be tested
    Example:
    - `npm install`
    - Go to '...'
    - Click on '....'
    - Scroll down to '....'
    - See error
 -->
Open the vercel deployment and test adding/editing/deleting comments as different users.

## Checklist
<!-- Before submitting a PR, address each item -->

- [x] **Quality**: This PR builds and passes the respective npm run test and works locally
- [ ] **Tests**: This PR includes thorough tests
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

chrispinkney commented on web/src/components/List/MovieListGrid/MovieListComments/MovieListComment.tsx:
```diff
@@ -0,0 +1,67 @@
+	const enableEditing = user.username == comment.username;
+
+	const commentUpdated = comment.createdAt != comment.updatedAt;
+	const formattedDate = new Date(comment.createdAt).toLocaleString('en-CA', {
```
I think you can set this value to [`Navigator.language`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) to use the user's browser setting instead of hardcoding it to en-ca for all users?

Thinbishop replied:
Implemented!

chrispinkney reviewed (changes requested):
![image](https://github.com/chrispinkney/cinesync/assets/7242003/b5a0a1a4-c31f-4bc3-b7f6-aba8a77dc2e4)

- When adding a new comment the input field is just one line long, can we make it a text area so we can add longer comments or another sentence? Not sure if the backend supports newline characters though?
- Also there doesn't seem to be debouncing for adding a comment, I hit shift+end when trying to see if I could add a new line and it added the same comment three times.
- Oh and I can add just a space as a comment 😂
- Also a list creator has authorization to delete any comment on a list that they own, I don't think the PR allows for that.
- I think a confirmation for deleting a comment would be a good idea too, whether it's a dialog box opens or another icon that appears or something? Just an idea.

Thinbishop commented:
> * When adding a new comment the input field is just one line long, can we make it a text area so we can add longer comments or another sentence? Not sure if the backend supports newline characters though?

This has been implemented. The user will now be able to press enter to add a new line and needs to press Ctrl+Enter or click the check to submit.

> * Also there doesn't seem to be debouncing for adding a comment, I hit shift+end when trying to see if I could add a new line and it added the same comment three times.

I've modified it to clear the focus and set editing = false after submission to fix this issue.

> * Oh and I can add just a space as a comment 😂

Too bad, that's a valid comment... (we can add an issue for zod implementation if this is deemed a legitimate problem)

> * Also a list creator has authorization to delete any comment on a list that they own, I don't think the PR allows for that.

I've added this in. The list creator can now delete any comment.

> * I think a confirmation for deleting a comment would be a good idea too, whether it's a dialog box opens or another icon that appears or something? Just an idea.

I've added a new issue for this (and other delete buttons)
